### DISCUSSION TITLE
How to deploy with nodejs

### DISCUSSION THREAD

**Discussion by HerbertHe** [HerbertHe/iptv-sources#37] [Q&A] [answered]
How to deploy with nodejs

**Comment by HerbertHe** [comment, marked as answer]
The project is developed with nodejs, so you can deploy directly with following steps:

- Confirm that you have installed nodejs first. If not, you can install it with nvm or see the offical website <https://nodejs.org/en>
- Clone the repo.

```shell
git clone git@github.com:HerbertHe/iptv-sources.git

cd iptv-sources
```

- Install `yarn`

```shell
npm install yarn -g
```

- Install dependencies

```shell
yarn install
```

- Build scripts

```shell
yarn build
```

- Fetch sources

> If you want to use `environment variables`, you need to add them to PATH first. Or, run the following command with them.

```shell
yarn m3u
```

- Start static service

```shell
yarn serve
```

Then, you can use it with <http://127.0.0.1:8080> or <http://[::1]:8080>

**Reply by nekkoup** [reply]
在构建脚本时出错，请您帮帮我！
![微信截图_20240729181319](https://github.com/user-attachments/assets/acbcf33d-1641-4d6f-9410-4b918b1f0060)
